.github/workflows/link-checker.yaml

```diff
@@ -45,11 +29,11 @@ jobs:
         run: make install-site-deps
       - name: Build docs for link check
         run: make build-pages
-        # Using link-checker action to check generated HTML site
-      - name: Link Checker (generated site)
-        id: lc
-        uses: peter-evans/link-checker@v1
+        # Using link-checker action to check links in Markdown, HTML files
+      - name: Link Checker
+        uses: lycheeverse/lychee-action@768a13171aff975543be1d2f2dd740bc3dcd113c #v1.2.1
         with:
-          args: -d userdocs/site -r userdocs/site -x https://twitter.com|https://www.weave.works
-      - name: Fail if there were link errors
-        run: exit ${{ steps.lc.outputs.exit_code }}
+          fail: true
+          args: --exclude-all-private --exclude-file .github/workflows/exclude-file.txt --verbose --no-progress './**/*.md' './**/*.html'
+        env:
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

Comment:
why does the link checker need our github token?

Reply:
it's an optional thing to avoid rate limiting when checking Github links, but I have added a exclude on GitHub domain so we don't need this anymore